src/SFA.DAS.Funding.SystemAcceptanceTests/StepDefinitions/DoNotReleasePaymentsWhenPaymentsAreFrozenStepDefinitions.cs

```diff
@@ -0,0 +1,48 @@
+using SFA.DAS.Apprenticeships.Types;
+using SFA.DAS.Funding.SystemAcceptanceTests.Helpers;
+using SFA.DAS.Funding.SystemAcceptanceTests.TestSupport;
+using TechTalk.SpecFlow;
+
+namespace SFA.DAS.Funding.SystemAcceptanceTests.StepDefinitions
+{
+    [Binding]
+    public class DoNotReleasePaymentsWhenPaymentsAreFrozenStepDefinitions
+    {
+        private readonly ScenarioContext _context;
+        private readonly PaymentsFrozenEventHelper _paymentsFrozenEventHelper;
+        private PaymentsFrozenEvent _paymentsFrozenEvent;
+        private TestSupport.Payments[] _paymentEntity;
+
+        public DoNotReleasePaymentsWhenPaymentsAreFrozenStepDefinitions(ScenarioContext context)
+        {
+            _context = context;
+            _paymentsFrozenEventHelper = new PaymentsFrozenEventHelper(context);
+        }
+
+        [When(@"Employer has frozen provider payments")]
+        public async Task WhenEmployerHasFrozenProviderPayments()
+        {
+            _paymentsFrozenEvent = _paymentsFrozenEventHelper.CreatePaymentsFrozenMessage();
+
+            await _paymentsFrozenEventHelper.PublishPaymentsFrozenEvent(_paymentsFrozenEvent);
+        }
+
+        [Then(@"do not make an on-programme payment to the training provider for that apprentice")]
+        public async Task ThenDoNotMakeAnOn_ProgrammePaymentToTheTrainingProviderForThatApprentice()
+        {
+            var apiClient = new PaymentsEntityApiClient(_context);
+
+            bool paymentsStatus;
+
+            await WaitHelper.WaitForUnexpected(() =>
+            {
+                _paymentEntity = apiClient.GetPaymentsEntityModel().Model.Payments;
+
+                paymentsStatus = apiClient.GetPaymentsEntityModel().Model.PaymentsFrozen;
```

Comment:
I don't know how long it takes to .GetPaymentsEntityModel() but thinking about performance we don't need to do it twice:
_paymentEntity = apiClient.GetPaymentsEntityModel().Model; //(then where you need to access payments use _paymentEntity.Payments of course)
paymentsStatus = _paymentEntity.PaymentsFrozen; //(or just use this directly in your boolean statement)